File: 18258.py
from sys import stdin

q = []
N = int(input())
cmd = [[] for i in range(N)]

for i in range(N):
    cmd[i] = list(map(str, stdin.readline().split()))

#시간 초과는 list에서 pop할 때 요소를 지우지 않고 인덱스만 증가시키는 것으로 해결
front = 0
for i in range(N):
    if len(q)-front:
        if cmd[i][0]=="push":
            q.append(int(cmd[i][1]));
        elif cmd[i][0]=="pop":
            print(q[front])
            front = front+1
        elif cmd[i][0]=="size":
            print(len(q)-front)
        elif cmd[i][0]=="empty":
            print(0)
        elif cmd[i][0]=="front":
            print(q[front])
        elif cmd[i][0]=="back":
            print(q[-1])
    else:
        if cmd[i][0]=="push":
            q.append(int(cmd[i][1]));
        elif cmd[i][0]=="pop":
            print(-1)
        elif cmd[i][0]=="size":
            print(0)
        elif cmd[i][0]=="empty":
            print(1)
        elif cmd[i][0]=="front":
            print(-1)
        elif cmd[i][0]=="back":
            print(-1)

File: POSCAT summer seminar/18258.cpp
#include <iostream>
#include <queue>
#include <sstream>

using namespace std;

int main(){
    queue<int> q;
    string cmd;
    int N;

    cin>>N;
    getline(cin, cmd);
    for(int i =0;i<N;i++){
        getline(cin, cmd);
        istringstream ss(cmd);
        ss>>cmd;
        if(cmd=="push"){
            int n;
            ss>>n;
            q.push(n);
        }
        else if(cmd=="pop"){
            if(q.empty()) cout<<-1<<endl;
            else {
                cout<<q.front()<<endl;
                q.pop();}
        }
        else if(cmd=="size"){
            cout<<q.size()<<endl;
        }
        else if(cmd=="empty"){
            cout<<q.empty()<<endl;
        }
        else if(cmd=="front"){
            if(q.empty()) cout<<-1<<endl;
            else cout<<q.front()<<endl;
        }
        else if(cmd=="back"){
            if(q.empty()) cout<<-1<<endl;
            else cout<<q.back()<<endl;
        }
    }
}